src/main/java/de/srendi/advancedperipherals/common/util/LuaConverter.java

```diff
@@ -236,6 +309,69 @@ public static Map<String, Object> fluidToObject(@NotNull Fluid fluid) {
         return map;
     }
 
+    public static Map<String, Object> positionToObject(@NotNull Position position) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("x", position.x());
+        map.put("y", position.y());
+        map.put("z", position.z());
+        return map;
+    }
+
+    public static Map<String, Object> damageSourceToObject(@NotNull DamageSource damageSource) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("msgId", damageSource.getMsgId());
+        map.put("entity", entityToLuaUUID(damageSource.getEntity()));
+        map.put("directEntity", entityToLuaUUID(damageSource.getDirectEntity()));
+        map.put("foodExhaustion", damageSource.getFoodExhaustion());
+        map.put("sourcePosition", ObjectUtil.nullableValue(damageSource.getSourcePosition(), LuaConverter::positionToObject));
+        return map;
+    }
+
+    public static String mobTypeToString(@NotNull MobType mobType) {
+        if (mobType == MobType.UNDEFINED) {
+            return "undefined";
+        } else if (mobType == MobType.UNDEAD) {
+            return "undead";
+        } else if (mobType == MobType.ARTHROPOD) {
+            return "arthropod";
+        } else if (mobType == MobType.ILLAGER) {
+            return "illager";
+        } else if (mobType == MobType.WATER) {
+            return "water";
+        }
+        throw new IllegalArgumentException("Unrecognized MobType");
+    }
+
+    public static Map<String, Object> aabbToObject(@NotNull AABB aabb) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("minX", aabb.minX);
+        map.put("minY", aabb.minY);
+        map.put("minZ", aabb.minZ);
+        map.put("maxX", aabb.maxX);
+        map.put("maxY", aabb.maxY);
+        map.put("maxZ", aabb.maxZ);
+        return map;
+    }
+
+    public static Map<Integer, Object> uuidToLua(UUID uuid) {
+        int[] ints = UUIDUtil.uuidToIntArray(uuid);
+        Map<Integer, Object> data = new HashMap<>();
+        for (int i = 0; i < ints.length; i++) {
+            data.put(i + 1, ints[i]);
+        }
+        return data;
+    }
+
+    public static UUID luaToUUID(Map<?, ?> uuidList) {
+        int msb1 = (int) (double) uuidList.get(1.);
+        int msb2 = (int) (double) uuidList.get(2.);
+        int lsb1 = (int) (double) uuidList.get(3.);
+        int lsb2 = (int) (double) uuidList.get(4.);
+        long msb = (((long) msb1) << 32) | (msb2 & 0xffffffffL);
+        long lsb = (((long) lsb1) << 32) | (lsb2 & 0xffffffffL);
+        return new UUID(msb, lsb);
+    }
```

Comment:
Well let's be easy here, use string but not int list

Reply:
Getting the real UUID from the UUID string is _slow_. I had noticeable slowdowns during testing because of it. You don't really need to deal with the UUID in Lua land, you just move it around and call methods with it, so it doesn't really matter how it is represented.

Reply:
> Getting the real UUID from the UUID string is slow.
I have no idea why parse a UUID string will be slow, UUID basically just a hex sequence, and it should be really fast. But I'll do some test and benchmark later.

> You don't really need to deal with the UUID in Lua land
That's false, sometimes player may want store entities in a table, and use UUID as a key.

Reply:
I didn't think it would be slow either. There is a chance that the issue was something else and I optimized in the wrong spot here. And thats true, I didnt think of using UUID as a table key.

Reply:
I just did some benchmark, `UUID.fromString` use average 20ns/op, and parse from map use `10ns/op`.
parse from string is twice slower, but still only slower 10ns, which should not be a big deal I believe

Reply:
Oh it must have been something else then. I'll change it back.